presto-main/src/main/java/com/facebook/presto/server/security/PasswordAuthenticator.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import com.facebook.presto.spi.security.AccessDeniedException;
+import com.google.common.base.Splitter;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import java.security.Principal;
+import java.util.Base64;
+import java.util.List;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static java.util.Objects.requireNonNull;
+
+public class PasswordAuthenticator
```

Comment:
I am not huge fan of the name of this class give that:
 * SPI contains interface with the same name
 * that interface is used by this class.

Reply:
I agree, though I couldn't think of a better name. Fortunately, the SPI interface is used implicitly in this class, so it doesn't require a fully qualified name. I was thinking of `DelegatingPasswordAuthenticator`, though I'm not sure that's an improvement.

Reply:
I don't have better name neither. :/ It does not seem `DPA` would be better as suffix is still a bit confusing. Not a big issue. Leave it please as we do not see alternative.

Reply:
How about `SystemPasswordAuthenticator`?

Reply:
Or maybe `BasicAuthenticator` since this handles basic auth?